src/libsystemd/sd-device/sd-device.c

```diff
@@ -152,6 +152,7 @@ int device_set_syspath(sd_device *device, const char *_syspath, bool verify) {
                 if (r < 0)
                         return log_debug_errno(r, "sd-device: Failed to get target of '%s': %m", _syspath);
 
+                path_simplify(syspath, false);
```

Comment:
hmm, why did chase_symlinks() return a non-simplified path? that's surprising. I would assume we should always return a simplified path, hence simpliying the result of chase_symlinks() should never be necessary, and always be a NOP?

Reply:
I will check `chase_symlinks()`, but before this PR,
```
$ udevadm info --path ///devices/virtual/workqueue/.///writeback
P: ////devices/virtual/workqueue///writeback
L: 0
E: DEVPATH=////devices/virtual/workqueue///writeback
E: SUBSYSTEM=workqueue
```
and after this PR,
```
$ udevadm info --path ///devices/virtual/workqueue/.///writeback
P: /devices/virtual/workqueue/writeback
L: 0
E: DEVPATH=/devices/virtual/workqueue/writeback
E: SUBSYSTEM=workqueue
```
So, it seems `chase_symlinks()` simplifies only dots, not slashes.

Reply:
we should fix that. this looks like a bug in chase_symlinks().

we should only return perfectly simplified paths in chase_symlinks() i am sure.